ext/web/08_text_encoding.js

```diff
@@ -14,27 +14,52 @@ const ops = core.ops;
 import * as webidl from "ext:deno_webidl/00_webidl.js";
 import { createFilteredInspectProxy } from "ext:deno_console/01_console.js";
 const {
+  ArrayBufferIsView,
   DataViewPrototypeGetBuffer,
   DataViewPrototypeGetByteLength,
   DataViewPrototypeGetByteOffset,
+  FunctionPrototypeCall,
+  ObjectGetOwnPropertyDescriptor,
+  ObjectPrototypeIsPrototypeOf,
   PromiseReject,
   PromiseResolve,
   // TODO(lucacasonato): add SharedArrayBuffer to primordials
-  // SharedArrayBufferPrototype
+  // SharedArrayBufferPrototype,
   StringPrototypeCharCodeAt,
   StringPrototypeSlice,
   SymbolFor,
-  TypedArrayPrototypeSubarray,
   TypedArrayPrototypeGetBuffer,
   TypedArrayPrototypeGetByteLength,
   TypedArrayPrototypeGetByteOffset,
   TypedArrayPrototypeGetSymbolToStringTag,
-  Uint8Array,
-  ObjectPrototypeIsPrototypeOf,
-  ArrayBufferIsView,
+  TypedArrayPrototypeSubarray,
   Uint32Array,
+  Uint8Array,
 } = primordials;
 
+// https://tc39.es/ecma262/#sec-get-sharedarraybuffer.prototype.bytelength
+let _getSharedArrayBufferByteLength;
+
+function getSharedArrayBufferByteLength(value) {
+  // TODO(kt3k): add SharedArrayBuffer to primordials
+  _getSharedArrayBufferByteLength ??= ObjectGetOwnPropertyDescriptor(
+    // deno-lint-ignore prefer-primordials
+    SharedArrayBuffer.prototype,
+    "byteLength",
+  ).get;
+
+  return FunctionPrototypeCall(_getSharedArrayBufferByteLength, value);
+}
+
+function isSharedArrayBuffer(value) {
+  try {
+    getSharedArrayBufferByteLength(value);
+    return true;
+  } catch {
+    return false;
+  }
+}
+
```

Comment:
Doing this across the codebase will be bad for perf and might also cause fast calls to deoptimize.

Reply:
Would it be faster to introduce `op_is_shared_array_buffer`?

Reply:
Yes, that would be better 👍